Sources/Fluent/Query/QueryField.swift

```diff
@@ -19,11 +19,15 @@ public struct QueryField: Hashable {
 
     /// The name of the field.
     public var name: String
+    
+    /// Optional alias for the field.
+    public var alias: String?
 
     /// Create a new query field.
-    public init(entity: String? = nil, name: String) {
+    public init(entity: String? = nil, name: String, alias: String? = nil) {
```

Comment:
I think aliases might be SQL-specific. We need to do some research to see how other DBs handle this. This may call for `QueryField` to be broken into some additional types. `QueryField` is used right now to represent data coming _back_ from the database. That data should never specify an alias, so it would be confusing to use this type with optional alias there. That should only be used when generating queries.